File: src/test/scala/com/exasol/cloudetl/scriptclasses/ImportPathTest.scala
package com.exasol.cloudetl.scriptclasses

import scala.jdk.CollectionConverters._

import org.mockito.Mockito._

class ImportPathTest extends PathTest {

  test("generateSqlForImportSpec throws with message to the new scriptclass name") {
    when(metadata.getScriptSchema()).thenReturn(schema)
    when(importSpec.getParameters()).thenReturn(properties.asJava)
    val thrown = intercept[IllegalArgumentException] {
      ImportPath.generateSqlForImportSpec(metadata, importSpec)
    }
    assert(thrown.getMessage().contains("Please use the FilesImportQueryGenerator"))
  }

}

File: src/test/java/com/exasol/cloudetl/scriptclasses/ImportPathTest.java
package com.exasol.cloudetl.scriptclasses;

import static org.junit.jupiter.api.Assertions.*;
import static org.mockito.Mockito.*;

import org.junit.jupiter.api.Test;

class ImportPathTest extends PathTest {
    @Test
    void generateSqlForImportSpecThrowsWithMessageToTheNewScriptclassName() {
        when(this.metadata.getScriptSchema()).thenReturn(this.schema);
        when(this.importSpec.getParameters()).thenReturn(this.properties);
        final IllegalArgumentException thrown = assertThrows(IllegalArgumentException.class,
                () -> ImportPath.generateSqlForImportSpec(this.metadata, this.importSpec));
        assertTrue(thrown.getMessage().contains("Please use the FilesImportQueryGenerator"));
    }
}
